validator/validator.bal

```diff
@@ -17,6 +17,78 @@
 import ballerina/constraint;
 import ballerinax/health.fhir.r4;
 import ballerinax/health.fhir.r4.parser;
+import ballerina/http;
+import ballerina/io;
+import ballerina/lang.regexp;
+
+# Description.
+#
+# + detailedErrors - User friendly error messages
+public type FHIRValidationIssueDetail record {
+    *r4:FHIRIssueDetail;
+    string[]? detailedErrors =();
+};
+
+# Utility function to create FHIRError.
+#
+# + message - Message to be added to the error
+# + errServerity - serverity of the error
+# + code - error code
+# + diagnostic - (optional) diagnostic message
+# + expression - (optional) FHIR Path expression to the error location
+# + cause - (optional) original error
+# + errorType - (optional) type of the error
+# + httpStatusCode - (optional) [default: 500] HTTP status code to return to the client
+# + return - Return Value Description
+# + parsedErrors - (optional) usefriendly error messages parsed from original error message
+public isolated function createFHIRError(string message, r4:Severity errServerity, r4:IssueType code,
+        string? diagnostic = (), string[]? expression = (), error? cause = (),
+        r4:FHIRErrorTypes? errorType = (), string[]? parsedErrors = (),  int httpStatusCode = http:STATUS_INTERNAL_SERVER_ERROR)
+        returns r4:FHIRError {
+    string diagnosticMessage = diagnostic != () ? string `${message} due to ${diagnostic}` : message;
+    string[] detailedErrors = parsedErrors != () ? parsedErrors : [diagnosticMessage];
+    return createTypedError(diagnosticMessage, errServerity, code, diagnostic, expression, cause, errorType,
+                                internal = false, httpStatusCode = httpStatusCode, detailedErrors = detailedErrors );
+}
+isolated function createTypedError(string message, r4:Severity errServerity, r4:IssueType code,
```

Comment:
we can merge these two methods since `createTypedError` is not being used in any other place.